src/js/binary.js

```diff
@@ -29,6 +29,109 @@ export function navMenuListener() {
+            // create list
+            $list = $('<ul />', { class: 'select-options' }).insertAfter($select_dropdown);
+        }
+
+        const optgroups = $(selector).children('optgroup');
+        if (optgroups.length) {
+            // break down group into labels with its list items
+            optgroups.each((idx, el) => {
+                const options = $(el).children();
+                const label   = $(el).attr('label');
+                appendToList(options, label);
+            });
+        } else {
+            const options = $(selector).children('option');
+            appendToList(options);
+        }
+
+        // Attach event listeners
+        $select_dropdown.off('click').on('click', (e) => {
+            e.stopPropagation();
+            // expand dropdown expand/collapse
+            const $siblings = $('.select-dropdown').not(e.target);
+            if ($siblings.hasClass('show')) {
+                $siblings.removeClass('show');
+            }
+            $(e.target).toggleClass('show');
+        });
+
+        $list_items = $list.children('li');
+        $list_items.off('click').on('click', (e) => {
+            e.stopPropagation();
+            const target = e.target;
+            if (e.target.classList.contains('disabled')) return;
+            $select_dropdown.text($(target).text()).removeClass('show').wrapInner('<span></span>');;
+
+            const selected_value = $(selector).val();
+            const dropdown_value = $(target).attr('value');
+
+            // sync original select with selected dropdown value
+            if (selected_value !== dropdown_value) {
+                const event = new Event('change');
+                // dispatch event to trigger onChange value
+                $(selector).val(dropdown_value).get(0).dispatchEvent(event);
+                $list_items.not(target).each((idx, el) => {
+                    $(el).removeClass('selected');
+                });
+                $(target).addClass('selected');
+            }
+        });
+
+        // collapse dropdown when clicking outside
+        $(document).click((e) => {
+            if ((!$list_items.is(e.target) && !$list_items.has(e.target).length)
+                 && $select_dropdown.hasClass('show')) {
+                $select_dropdown.removeClass('show');
+            }
+        });
+    };
+
+    function appendToList(options, label) {
+        if (has_label && label) {
+            $('<li />', {
+                text    : label,
+                addClass: 'select-items label',
+            }).appendTo($list);
+        }
+
+        $.map(options, (el) => {
+            const is_disabled = $(el).is(':disabled');
+            const is_selected = $(el).is(':selected');
+            const className   = `select-items${is_selected ? ' selected': ''}${is_disabled ? ' disabled': ''}`;
+            $('<li />', {
+                text    : $(el).text(),
+                value   : $(el).val(),
+                addClass: className,
```

Comment:
this should also be able to just remain as `class:`, unless I'm missing something